### PR TITLE
Grab xml labels

### PR DESCRIPTION
**Pull request by wsf246**
**Problem:** 
Pubmed Abstracts with labels are not pulled in correctly
**Solution:** 
Use nokogiri to pull out labels and add it to the text

### CODE REVIEW

**Comment by wsf246** [comment]
image

**Comment by wsf246** [comment]
dunno if the way I did the formatting was the best, but is a significant improvement over what it was pulling before.  

You can use this pubmed ID to test 19675114

File: app/views/papers/_tablist.html.erb [hunk 1]
```diff
@@ -19,7 +19,7 @@
   <% end %>
 </ul>
 <div class="collapse paper-detail well" id="abstract-<%= reference.id %>">
-  <p><%= paper.abstract %></p>
+  <p><%= simple_format(paper.abstract) %></p>
```
Comment (briankung) [inline app/views/papers/_tablist.html.erb:22]: This is cool

File: app/models/services/pubmed.rb [hunk 2]
```diff
@@ -38,15 +38,10 @@ def get_uid_metadata(uid)
...
-      'PubmedArticleSet',
-      'PubmedArticle',
-      'MedlineCitation',
-      'Article',
-      'Abstract',
-      'AbstractText'
-    )
+    result = Nokogiri::XML(response)
+    abstract_xml = result.xpath("//Abstract//AbstractText")
+    abstract = abstract_xml.map {|a| "#{a['Label'].humanize + ':' if a['Label']} #{a.text}"}.join("\n \n")
```
Comment (briankung) [inline app/models/services/pubmed.rb]: I like this, but I'm gonna make a few tweaks

File: app/models/services/pubmed.rb [hunk 3]
```diff
@@ -55,7 +50,6 @@ def get_abstract(uid)
        abstract = pubmed_scrape[/#{"AbstractText"}(.*?)#{"</AbstractText>"}/m, 1].partition('>').last
       end
     end
-    abstract = abstract.join ' ' if abstract.respond_to? :join
```
Comment (briankung) [inline app/models/services/pubmed.rb:58]: 👏 👏 👏 deleting code👏 👏 👏

File: app/models/services/pubmed.rb [hunk 4]
```diff
@@ -38,15 +38,10 @@ def get_uid_metadata(uid)
...
     response = Net::HTTP.get(abstract_uri)
-    result = Hash.from_xml(response)
-    abstract = result.dig(
-      'PubmedArticleSet',
-      'PubmedArticle',
-      'MedlineCitation',
-      'Article',
-      'Abstract',
-      'AbstractText'
-    )
```
Comment (briankung) [inline app/models/services/pubmed.rb:49]: 👏 👏 👏 deleting code👏 👏 👏

**Review by briankung — CHANGES_REQUESTED** [review]
Looks good! I'm gonna make a tweak and then approve it.

File: app/models/services/pubmed.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 class Pubmed
   def initialize(options={})
-    @base_url = 'http://eutils.ncbi.nlm.nih.gov'
+    @base_url = 'https://eutils.ncbi.nlm.nih.gov'
```
Comment (briankung) [inline app/models/services/pubmed.rb:3]: 👏 👏 👏 HTTPS👏 👏 👏

**Review by briankung — APPROVED** [review]

**Comment by briankung** [comment]
Still works with my changes, so this is 👍  to merge. I'm gonna be on the move for a bit so feel free to

**Pull request merged**